60 format option to match buf or rvls

UmeSiyah opened the pull request:
Hello,

I have renamed the branch to remove the "WIP" prefix, as my work is now ready for review. However, this action closed the existing pull request, so I am opening a new one.

Thank you.

doubleailes commented:
I have trig the CI/Cd and the performance have drop a lot if you check [this CI](https://github.com/forticheprod/fls/actions/runs/14572259719/job/40925090322) on your current branch run the benchmark in **3.3210 µs** for the *Parsing/Mono/1*, instead of **1.3858 µs** for [a Ci running main branch](https://github.com/forticheprod/fls/actions/runs/14595411721/job/40939998369).
Can you try to reduce the drop of performance?

UmeSiyah commented:
I can try! However, I haven't optimized Rust code before. Do you have any recommendations for a Rust profiler I could use?

UmeSiyah commented:
Hello,

I'm having trouble identifying the source of the slowdowns. Using Flamegraph, I identified and removed an unnecessary hashmap usage (which corresponds to my latest commits), which improved the performance from **6.7953 µs** to **5.5447 µs** (for the _Parsing/Mono/1_ on my machine). However, I'm now stuck and struggling to identify further optimizations.

I've added a `parse` benchmark to test the `parse_dir` function in isolation, but I haven't noticed any difference (all my tests remain at 1.4 ms).

Here are the two Flamegraphs if you can take a look (I didn't find a tool for comparison). To generate them, I navigate to the `samples/small` directory and run:
```cargo flamegraph --flamechart --inverted -o ../../main.svg```

![60](https://github.com/user-attachments/assets/32e6c133-ce36-4844-88c3-f4f437cca2ce)
![main](https://github.com/user-attachments/assets/d49f0625-206c-4529-aaea-e5382823ac60)

UmeSiyah commented:
I tested this morning, and it's `strfmt` that's slowing things down. I wanted to use this cargo to in the future allow the user to define their own template string to format the output, but due to the performance loss, I'm wondering if I should rewrite otherwise. What do you think?
